content/en/docs/setup/getting-started/index.md

```diff
@@ -72,142 +56,341 @@ Download the Istio release which includes installation files, samples, and the
-traffic; HTTP/1.0 is not supported.
+In certain environments, the load balancer may be exposed using a host name, instead of an IP address.
+In this case, the ingress gateway's `EXTERNAL-IP` value will not be an IP address,
+but rather a host name, and the above command will have failed to set the `INGRESS_HOST` environment variable.
+Use the following command to correct the `INGRESS_HOST` value:
+
+{{< text bash >}}
+$ export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+{{< /text >}}
+
 {{< /warning >}}
 
+**Follow these instructions if your environment does not have an external load balancer and choose a node port instead.**
+
+Set the ingress ports:
+
+{{< text bash >}}
+$ export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+$ export SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+{{< /text >}}
+
+_GKE:_
+
+{{< text bash >}}
+$ export INGRESS_HOST=<workerNodeAddress>
+{{< /text >}}
+
+You need to create firewall rules to allow the TCP traffic to the `ingressgateway` service's ports.
+Run the following commands to allow the traffic for the HTTP port, the secure port (HTTPS) or both:
+
 {{< text bash >}}
-$ kubectl label namespace <namespace> istio-injection=enabled
+$ gcloud compute firewall-rules create allow-gateway-http --allow tcp:$INGRESS_PORT
+$ gcloud compute firewall-rules create allow-gateway-https --allow tcp:$SECURE_INGRESS_PORT
 {{< /text >}}
 
-Alternatively, you can manually inject Envoy containers in your application pods before deploying them, using
-[`istioctl kube-inject`](/docs/reference/commands/istioctl/#istioctl-kube-inject):
+_Docker For Desktop:_
 
 {{< text bash >}}
-$ istioctl kube-inject -f <your-app-spec>.yaml | kubectl apply -f -
+$ export INGRESS_HOST=127.0.0.1
 {{< /text >}}
 
-## Deploy an application or the Bookinfo sample {#Bookinfo}
+_Other environments (e.g., IBM Cloud Private, etc.):_
+
+{{< text bash >}}
+$ export INGRESS_HOST=$(kubectl get po -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].status.hostIP}')
+{{< /text >}}
+
+{{< /tab >}}
+
+{{< /tabset >}}
 
-If you have an application ready to go, deploy it:
+1.  Set `GATEWAY_URL`:
+
+    {{< text bash >}}
+    $ export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
+    {{< /text >}}
+
+1.  Ensure an IP address and port were successfully assigned to the environment variable:
+
+    {{< text bash >}}
+    $ echo $GATEWAY_URL
+    192.168.99.100:32194
+    {{< /text >}}
+
+### Verify external access {#confirm}
+
+Confirm that the Bookinfo application is accessible from outside. Copy the
+output of this command and paste into your browser:
 
 {{< text bash >}}
-$ kubectl create -n <namespace> -f <your-app-spec>.yaml
+$ echo http://$GATEWAY_URL/productpage
```

Comment:
Is this actually valid thing to put into browser?
`GATEWAY_URL` will have internal node port set (here: `32194`).
Yet output from:
```sh
kubectl get svc istio-ingressgateway -n istio-system
```
shows that this is **destination** port for request that will originate **to** `http://${INGRESS_HOST}:80` (!)
The former one (with port `80`) works for me while testing. The one with `nodePort` gives no responses at all - from local browser.

Reply:
Hi pazkooda! Please raise this as an issue, rather than a comment on this merged PR, so it can be triaged and looked at.  Thanks!